cryptofeed/exchanges/binance.py

```diff
@@ -287,36 +322,140 @@ async def _snapshot(self, pair: str) -> None:
         url = f'{self.rest_endpoint}/depth?symbol={pair}&limit={max_depth}'
         resp = await self.http_conn.read(url)
         resp = json.loads(resp, parse_float=Decimal)
+        return resp
+
+    async def _reset_snapshot(self, pair: str) -> None:
+        resp = await self._fetch_snapshot(pair)
+        receipt_timestamp = time.time()
 
         std_pair = self.exchange_symbol_to_std_symbol(pair)
         self.last_update_id[std_pair] = resp['lastUpdateId']
         self._l2_book[std_pair] = OrderBook(self.id, std_pair, max_depth=self.max_depth, bids={Decimal(u[0]): Decimal(u[1]) for u in resp['bids']}, asks={Decimal(u[0]): Decimal(u[1]) for u in resp['asks']})
+        self._l2_book[std_pair].sequence_number = resp['lastUpdateId']
+        self._l2_book[std_pair].raw = resp
+        
+        if self.refresh_snapshot:
+            # Delete buffer and reset time since snapshot update
+            self._next_snapshot_time[std_pair] = receipt_timestamp + self._next_snapshot_refresh_interval()
+            self._fetch_snapshop_buffer.pop(std_pair, None)
         ts = self.timestamp_normalize(resp['E']) if 'E' in resp else None
-        await self.book_callback(L2_BOOK, self._l2_book[std_pair], time.time(), raw=resp, timestamp=ts, sequence_number=resp['lastUpdateId'])
```

Comment:
setting them above wont work, book_callback will reset them to None. They need to be passed in like they were previously

Reply:
Makes sense